earth_enterprise/src/fusion/autoingest/plugins.src/PacketGen.src

```diff
@@ -122,21 +124,23 @@ void PacketGenAssetVersionImplD::UpdateChildren(
     // Determine which insets intersect us at this level
     const khLevelCoverage levCov(config.coverage.levelCoverage(level));
     std::vector<uint> neededIndexes;
+
     neededIndexes.reserve(numInsets);
-    if (type == AssetDefs::Imagery) {
-      FindNeededImageryInsets(khInsetCoverage(levCov),
-                              extra.insetInfo, numInsets,
-                              neededIndexes,
-                              config.beginMinifyLevel,
-                              config.endMinifyLevel);
-    } else {
-      FindNeededTerrainInsets(khInsetCoverage(levCov),
-                              extra.insetInfo, numInsets,
-                              neededIndexes,
-                              config.beginMinifyLevel,
-                              config.endMinifyLevel);
+    auto gencov = khInsetCoverage(levCov);
+    overlapEnvelope<ProductAssetVersion> env(type, gencov,
+                         extra.insetInfo, extra.insetInfo.size(),
+                         config.beginMinifyLevel, config.endMinifyLevel);
+
+    // there are situations where this can happen and must be accounted for
+    if (calc == nullptr)
+    {
+        calc.reset(new OverlapCalculator<ProductAssetVersion>(env));
+        calc->PreprocessForInset();
     }
 
+    neededIndexes = std::move(calc->GetOverlapForLevel());
```

Comment:
I believe the level needs to be passed as a parameter here. If I'm not mistaken that's one of the important pieces of information for figuring out the overlap for each level.

Reply:
unless i misunderstand it, is already being covered within khinsetcoverage, but i am passing it down either way